### PR TITLE
wip: switch to protobuf types

### PR DESCRIPTION
**Pull request by zonotope**
This is a draft pull request that refactors the gossip4 types into messages so that we can use them to validate receive token transactions. What's left to do here is to finish wrapping the Checkpoint type to cache the cbor representation within the app, and then flesh out the proof validater

### CODE REVIEW

File: gossip/node.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	logging "github.com/ipfs/go-log"
 	"github.com/libp2p/go-libp2p-core/network"
 	"github.com/quorumcontrol/chaintree/nodestore"
+	"github.com/quorumcontrol/messages/build/go/gossip"
```
Comment (cap10morgan) [inline gossip/node.go]: Shouldn't this be in `v2`?

Comment (zonotope) [reply gossip/node.go]: why yes, yes it should!

**Comment by zonotope** [comment]
I've finished the wrapping and added a function to validate proofs. There are some broken tests that I'll have to fix, as well as replacing the block validators to use the new `validateProof` function, as well as tests around that.

**Comment by zonotope** [comment]
:ping_pong:

**Comment by tobowers** [comment]
does the pong emoji mean this is ready for review?

**Comment by zonotope** [comment]
yes. i guess notifications for formerly draft pull requests don't work. i will more explicitly ping next time.

**Pull request merged**